### PR TITLE
Make resource compatible with new policy

### LINKED ISSUE #422: get_policy crashes with minio server
```
Traceback (most recent call last):
  File "tests.py", line 188, in <module>
    main()
  File "tests.py", line 162, in main
    policy_name = client.get_bucket_policy(bucket_name)
  File "/home/harsha/repos/minio-py/minio/api.py", line 341, in get_bucket_policy
    return policy.get_policy(statements, bucket_name, prefix)
  File "/home/harsha/repos/minio-py/minio/policy.py", line 463, in get_policy
    fnmatch.fnmatch(object_resource, resource)):
  File "/usr/lib/python2.7/fnmatch.py", line 43, in fnmatch
    return fnmatchcase(name, pat)
  File "/usr/lib/python2.7/fnmatch.py", line 77, in fnmatchcase
    re_pat = _cache[pat]
TypeError: unhashable type: 'list'
```

### PR DESCRIPTION
**Pull request by ChillarAnand**
closes #422

### CODE REVIEW

File: minio/policy.py [hunk 1]
```diff
@@ -477,23 +477,24 @@ def get_policy(statements, bucket_name, prefix=''):
...
-                matched_resource = resource
-        elif resource == bucket_resource:
-            common_found, read_only, write_only = _get_bucket_policy(s, prefix)
-            bucket_common_found = bucket_common_found or common_found
-            bucket_read_only = bucket_read_only or read_only
-            bucket_write_only = bucket_write_only or write_only
+        resources = _get_resource(s)
+        for resource in resources:
+            if (resource == object_resource or
+                    fnmatch.fnmatch(object_resource, resource)):
```
Comment (balamurugana) [inline minio/policy.py]: 1. Have a function to work with single resource and use that function in a for loop.
2. Write a unit test for this case.  You could refer `minio/minio-go` for unit tests.

**Comment by harshavardhana** [comment]
Are there any changes pending on this PR?

File: minio/policy.py (hunk 1 above)
Comment (ChillarAnand) [inline minio/policy.py]: > Have a function to work with single resource and use that function in a for loop.

That function needs a lot of context that needs to be passed. Won't that make ugly?

@harshavardhana Yes, its still WIP.

**Comment by balamurugana** [comment]
> That function needs a lot of context that needs to be passed. Won't that make ugly?


Could you add more about `a lot of context`?

**Comment by ChillarAnand** [comment]
@balamurugana can you check this https://github.com/minio/minio-py/pull/456/commits/2db1e751389584103e03891bbc0183554229a063?

**Comment by harshavardhana** [comment]
ping @balamurugana - can you review..

**Comment by harshavardhana** [comment]
ping @balamurugana can you review?

**Pull request merged**